cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1329,6 +1337,19 @@ if(ARROW_USE_BOOST)
     endif()
     if(Boost_VERSION VERSION_GREATER_EQUAL 1.80)
       target_compile_definitions(Boost::process INTERFACE "BOOST_PROCESS_HAVE_V2")
+      if((NOT WIN32) AND (NOT ARROW_WITH_MUSL))
```

Comment:
Instead of repeating this, can we factor it out?

Reply:
How about https://github.com/apache/arrow/pull/45168/commits/209c60dcec1833e138ba55a1af976974ebc66c3d ?